### DISCUSSION TITLE
check dataset for suitability ASReview

### DISCUSSION THREAD

**Discussion by Rinus-Verdonschot-UB** [asreview/asreview#1916] [Ideas]
Dear group,

I have a suggestion, not sure if its good or not. Running several simulations these days I see that many datasets would have profited when done with ASReview, however, some would not have. Often the reason would be the heterogeneity of the dataset. For example, there could be several subsets listed under one big topic, which could then not cluster together well and if seeded improperly clusters might be recognized (much) later.

I think it would be very nice if we, a priori, could gauge whether a dataset would be suitable to review with ASReview.
One thing which might be considered, is to take a dataset, then initialize using SAFE (random 1% with min. 1 relevant) and do this a number of times (say N=5 times). Then we print out the ranking of each initialization and check for commonalities. If the dataset is quite homogeneous then similar articles should be in the top ranked articles (e.g. N=100) of each initialization. If the dataset is rather heterogenenous then different articles might enter the top 100. We might then calculate some correlation for this to obtain some metric of homo/heterogeneity of the dataset?

Maybe its not a great, idea, just wanted your opinion, maybe there are other things which can be done to assess a dataset's suitability a priori. 

Kind regards,

Rinus Verdonschot & Marieke Schor
University Library, Maastricht University

**Comment by Rensvandeschoot** [comment]
Yes, indeed, the performance varies across datasets _and_ over models; see [this scoping review](https://osf.io/preprints/psyarxiv/67zmt) by @jteijema, providing an overview of existing simulation studies. We also tested the performance of many models using the 26 labeled systematic review datasets of the [Synergy data collection](https://dataverse.nl/dataset.xhtml?persistentId=doi:10.34894/HE6NAQ). See our [pre-print](https://osf.io/preprints/psyarxiv/2w3rm) and especially Figure 3, in which you can see that for some datasets, and it does not matter what model or training data you use, the performance is always good (on the right side). However, for some datasets, it matters quite a lot (see the large variance for some datasets), and the performance is always lower for some data.  

![image](https://github.com/user-attachments/assets/d88022ac-1e60-4e42-af07-ad78310dc124)

We obtained similar results when testing the performance of one model on [14 different medical guidelines](10.1186/s13643-024-02590-5), and here we went back to the initial screeners and tried to explain why the performance was lower for some data than for other datasets; sometimes because of noisy labeling decisions by humans.

So, we need many more fully labeled datasets to start building a prediction model to explain the different performance levels. If you have such datasets available on, for example, OSF or other data repositories, we could add them to the Synergy collection via our pipeline available on [Github](https://github.com/asreview/synergy-dataset). 

With more of such dataset, you could also test your hypotheses! Feel free to already test it using the Synergy collection!
